src/Planners/include/BipedalLocomotion/Planners/QuinticSpline.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file QuinticSpline.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_QUINTIC_SPLINE_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_QUINTIC_SPLINE_H
+
+#include <vector>
+#include <memory>
+
+#include <Eigen/Dense>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+/**
+ * Quintic spline implement a 5-th order polynomial spline in \$f\mathbb{R}^n\$f.
+ */
+class QuinticSpline
```

Comment:
Actually I don't know, I would like to have a similar interface to an [`iDynTree::CubicSpline`](https://robotology.github.io/idyntree/master/classiDynTree_1_1CubicSpline.html). Furthermore I was thinking to add also a function to get the entire trajectory (from the initial to the final time instant) this may help to use this class also with predictive controllers.

Reply:
I don't think these things are mutually exclusive. Inheriting from ``Advanceable`` would imply having two additional methods, but you can still use it as you prefer.

Reply:
> I don't think these things are mutually exclusive. Inheriting from Advanceable would imply having two additional methods, but you can still use it as you prefer.

You are right, the only thing I don't like of advanceable in this case is the fact that you have to set the sampling time of the class (that could be not required) furthermore it is not really clear how the user should use it.
In other words:
```cpp
QuinticSpline sp;
// set the spline
sp.setSamplingTime(0.01)

auto traj = sp.get(); // <-------- Here I get the first value of the trajectory ( at t =  0 )
sp.advance(); // <----- Advace the trajectory
auto traj = sp.get(); // <-------- Here I get the second value of the trajectory ( at t = 0.01 )

sp.evaluatePoint(10, pos, vel, acc); // <-------- pos, vel and acc contains the state at t = 10

sp.advance(); // <----- Advace the trajectory
auto traj = sp.get(); // <-------- Here I get the second value of the trajectory ( at t = 0.02 )
```
I don't know if this may create problems in the future. If you think no I can add the possibility to set the sampling time. If it is set then the advance machinery works otherwise the user can get the trajectory only with `evaluatePoint()`.

Reply:
> You are right, the only thing I don't like of advanceable in this case is the fact that you have to set the sampling time of the class (that could be not required)

That is true. We can define a method like ``setAdvanceTimeStep``. In this way, it is clear that we are setting the time step for the ``advance`` method. We can discuss if this should be part of ``Advanceable`` or not. Regarding the example you showed, I don't think it is a big issue. ``Advanceable`` gives a basic interface for all the objects which output a trajectory. Then, if you have a reference to the actual class you can use its full-fledged interface.

Reply:
https://github.com/dic-iit/bipedal-locomotion-framework/pull/83/commits/02557d7c8b4ccfef3bc6d147cb4aa00283110274 introduces the advance capabilities https://github.com/dic-iit/bipedal-locomotion-framework/pull/83/commits/5eb51cdf3cb1de6aed87548c5ce0224ed07b3612 is the test.